spec/models/school_device_allocation_spec.rb

```diff
@@ -82,4 +82,43 @@
       end
     end
   end
+
+  context 'when in a virtual pool' do
+    let(:responsible_body) { create(:trust, :manages_centrally) }
+    let(:school) { create(:school, :with_preorder_information, :in_lockdown, responsible_body: responsible_body) }
+
+    subject(:allocation) { described_class.create!(device_type: 'std_device', cap: 100, devices_ordered: 87, allocation: 100, school: school) }
+
+    ### moving this into an around block causes test fails elsewhere in the suite?!?
+    before do
+      allocation
+      responsible_body.add_school_to_virtual_cap_pools!(school)
+      responsible_body.std_device_pool.update!(cap: 256, devices_ordered: 145)
+      allocation.reload
+
+      @original_state = FeatureFlag.active?(:virtual_caps)
```

Comment:
Al wrote a helper for this eg `context 'when authenticated', with_feature_flags: { mno_offer: 'active' }`

Reply:
Oh that sounds good. I was having issues with `around` blocks yesterday, for some reason it was causing unrelated tests in other parts of the suite to fail that worked fine when the around block wasn't used, there is also `FeatureFlag.temporarily_active` that takes a block but I took it all out when trying to fix the tests. But I'll look at that 👍